Index: source/icudefs.mk.in
===================================================================
--- source.orig/icudefs.mk.in
+++ source/icudefs.mk.in
@@ -131,6 +131,7 @@ DEFS = -DU_ATTRIBUTE_DEPRECATED=
 CFLAGS = @CFLAGS@
 # CXXFLAGS is for C++ only flags
 CXXFLAGS = @CXXFLAGS@
+CXXFLAGS+=-std=c++11
 # CPPFLAGS is for C Pre-Processor flags
 CPPFLAGS = @CPPFLAGS@
 # LIBCFLAGS are the flags for static and shared libraries.
